packages/flutter_tools/test/terminal_handler_test.dart

```diff
@@ -209,6 +220,74 @@ void main() {
       verify(mockResidentRunner.screenshot(mockFlutterDevice)).called(1);
     });
 
+    testUsingContext('r - hotReload supported and succeeds', () async {
+      when(mockResidentRunner.canHotReload).thenReturn(true);
+      when(mockResidentRunner.restart(fullRestart: false))
+          .thenAnswer((Invocation invocation) async {
+            return OperationResult(0, '');
+          });
+      await terminalHandler.processTerminalInput('r');
+
+      verify(mockResidentRunner.restart(fullRestart: false)).called(1);
+    });
+
+    testUsingContext('r - hotReload supported and fails', () async {
+      when(mockResidentRunner.canHotReload).thenReturn(true);
+      when(mockResidentRunner.restart(fullRestart: false))
```

Comment:
A test here where the restart throws to go along with the try {} catch {} requested above.